File: shoebill_plugins/LtrpApi/src/main/java/lt/ltrp/api/NamedEntityImpl.java
package lt.ltrp.api;

/**
 * @author Bebras
 *         2015.11.29.
 */
public class NamedEntityImpl extends EntityImpl {

    private String name;

    public NamedEntityImpl(int id, String name) {
        super(id);
        this.name = name;
    }

    public NamedEntityImpl(){
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: shoebill_plugins/LtrpApi/src/main/java/lt/ltrp/object/impl/NamedEntityImpl.kt
package lt.ltrp.`object`.impl;

import lt.ltrp.`object`.Entity
import lt.ltrp.`object`.NamedEntity

/**
 * @author Bebras
 *         2015.11.29.
 */
abstract  class NamedEntityImpl(uuid: Int, override var name: String): NamedEntity, EntityImpl(uuid) {

    constructor(): this(Entity.INVALID_ID, "invalid entity") {

    }

}
